package-res/resources/web/pentaho/data/filter/AbstractFilter.js

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../../lang/Base",
+  "./_Element",
+  "../TableView",
+  "../../util/arg",
+  "require"
+  //"./Or",
+  //"./And",
+  //"./Not",
+], function(Base, Element, TableView, arg, require,  Or, And, Not) {
+  "use strict";
+
+  /**
+   * @name AbstractFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @abstract
+   * @amd pentaho/data/filter/AbstractFilter
+   *
+   * @classdesc The `AbstractFilter` class is the abstract base class of
+   * classes that represent a filter.
+   *
+   * ### Remarks
+   *
+   * The following derived classes are also abstract:
+   *
+   * * {@link pentaho.data.filter.AbstractPropertyFilter}
+   * * {@link pentaho.data.filter.AbstractTreeFilter}
+   */
+  var AbstractFilter = Base.extend("pentaho.data.filter.AbstractFilter", /** @lends pentaho.data.filter.AbstractFilter# */{
+    /**
+     * Outputs a simple object that serializes the operation described by this filter.
+     * The syntax loosely follows the query language of MongoDB.
+     *
+     * @return {Object} Object.
+     */
+    toSpec: /* istanbul ignore next: placeholder method */ function() {
+      return null;
+    },
+
+    /**
+     * Tests if an entry is an element of the set defined by this filter.
+     *
+     * @param {pentaho.type.Element} - [dataTable]{@link pentaho.data.Table} entry.
+     * @return {boolean}
+     */
+    contains: /* istanbul ignore next: placeholder method */ function(entry) {
+      return false;
+    },
+
+    /**
+     * Returns a filter that is the inverse of this filter.
+     * @returns {*}
+     */
+    invert: function() {
+      if(!Not) Not = require("./Not");
+      return new Not(this);
+    },
+
+    /**
+     * Returns a filter that is the union of this filter with another.
+     * In other words, implements the OR operation between this filter and another.
+     * @param {} other -
+     * @returns {*}
+     */
+    or: function() {
```

Comment:
I don't follow. 

`filter.or(other)` is an `or` between `filter` and `other`. How can it merely return `other`
Or do you mean that I should have `filter.or() === filter`?